File: recyclerView/techsolpoint_recycler_view/app/src/main/java/com/example/techsolpoint_recycler_view/MainActivity.java
package com.example.techsolpoint_recycler_view;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    ArrayList<Model> models = new ArrayList<Model>();
    RecyclerView rvTechSolPoint;
    RvAdapter rvAdapter;
    TextView tvAdd, tvUpdate;
    EditText etEnterName;
    int position;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        rvTechSolPoint = findViewById(R.id.rv_list_item);
        tvAdd = findViewById(R.id.tv_add);
        etEnterName = findViewById(R.id.et_enter_name);
        tvUpdate = findViewById(R.id.tv_update);
        rvTechSolPoint.setHasFixedSize(true);
        LinearLayoutManager layoutManager = new LinearLayoutManager(getApplicationContext(), LinearLayoutManager.VERTICAL, false);
        rvTechSolPoint.setLayoutManager(layoutManager);
        rvAdapter = new RvAdapter(getApplicationContext(), models,
                new RvAdapter.Onclick() {
                    @Override
                    public void onEvent(Model model, int pos) {
                        position = pos;
                        tvUpdate.setVisibility(View.VISIBLE);
                        etEnterName.setText(model.getName());
                    }
                });
        rvTechSolPoint.setAdapter(rvAdapter);
        tvAdd.setOnClickListener(this);
        tvUpdate.setOnClickListener(this);
    }
    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.tv_add: {
                insertMethod(String.valueOf(etEnterName.getText()));
            }
            break;
            case R.id.tv_update: {
                models.get(position).setName(etEnterName.getText().toString());
                rvAdapter.notifyDataSetChanged();
                tvUpdate.setVisibility(View.GONE);
            }
            break;
        }
    }
    private void insertMethod(String name) {
        Gson gson = new Gson();
        try {
            JSONObject jsonObject = new JSONObject();
            jsonObject.put("name", name);
            Model model = gson.fromJson(String.valueOf(jsonObject), Model.class);
            models.add(model);
            rvAdapter.notifyDataSetChanged();
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }
}

File: recyclerView/Tap/app/src/main/java/com/example/tap/MainActivity.kt
package com.example.tap

import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity(), Clicked {
    val itemList = ArrayList<Data>()
    lateinit var rcv: RecyclerView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        fetchData()
        rcv = findViewById(R.id.rcv)
        //rcv.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        rcv.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        rcv.adapter = DataAdapter(itemList,this)
    }

    fun fetchData(){
        for (i in 1..6) {
            var obj = Data("Urban ", "X", false)
            itemList.add(obj)
        }
    }

   // override fun clickOnPosition(index: Int) {

//    }

    override fun clickOnPosition(index: Int) {
//        itemList[index].text ="ASASa"

        if (itemList[index].isSelected == false) {
            for(i in 0 .. itemList.size -1)
            {
                if(i!=index && itemList[i].isSelected==true)
                {
                    itemList[i].isSelected = false
                }
            }
            itemList[index].isSelected = true
            (rcv.adapter as DataAdapter).notifyDataSetChanged()
        }
        else if(itemList[index].isSelected == true)
        {
            itemList[index].isSelected = false
           // (rcv.adapter as DataAdapter).notifyItemChanged(index);
            (rcv.adapter as DataAdapter).notifyDataSetChanged()
        }
    }
}